CollAction/Frontend/app/project/UploadDescriptiveImage.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from "react";
+import UploadImage from "./UploadImage";
+import DropZone from "react-dropzone";
+import renderComponentIf from "../global/renderComponentIf";
+
+export default class UploadDescriptiveImage  extends UploadImage {
+    constructor(props: {}) {
+        super(props);
+
+        this.onDrop = this.onDrop.bind(this);
+        this.previewImage = this.previewImage.bind(this);
+        this.onRejected = this.onRejected.bind(this);
+        this.getFileInputElement = this.getFileInputElement.bind(this);
+        this.removeBanner = this.removeBanner.bind(this);
+    }
+
+    onDrop(accepted: File[], rejected: File[], event: any) {
+        super.loadImage(accepted, rejected, event);
+    }
+
+    previewImage(type: string, image: any) {
+        let background = document.getElementById("preview") as HTMLImageElement;
```

Comment:
This way of doing things feels like going against the point of react a bit. If you have type and image as part of your state, and then putting this in the render function, then you don't even need fildling with document.getElement, it's just a re-render (which should be fast with the virtual dom).

If image and type are part of the state

Reply:
I see what you mean; not sure if it's a good idea since images can take up a lot of memory. I'll profile it to see what making it reactive means for browser memory usage.

Reply:
Using reactive state now. Looks a lot better.

Reply:
And couldn't find a lot of difference in memory usage.